src/node/internal/internal_tls_wrap.ts

```diff
@@ -442,7 +443,35 @@ TLSSocket.prototype._finishInit = function _finishInit(this: TLSSocket): void {
 };
 
 TLSSocket.prototype._start = function _start(this: TLSSocket): void {
-  // Do nothing.
+  if (this.connecting) {
+    this.once('connect', this._start);
+    return;
+  }
+
+  // Socket was destroyed before the connection was established
+  if (this._handle == null) {
+    return;
+  }
+
+  // We first need to release the lock
+  this._handle.writer.releaseLock();
+  this._handle.reader.releaseLock();
+
+  try {
+    const socket = this._handle.socket.startTls();
+
+    this._handle = {
+      socket: socket,
+      writer: socket.writable.getWriter(),
+      reader: socket.readable.getReader({ mode: 'byob' }),
+      bytesRead: 0,
+      bytesWritten: 0,
+    };
+
+    startRead(this);
+  } catch (error) {
+    // Ignore errors.
```

Comment:
Hmmm... why ignore errors here? Should errors at this point instead cause the socket to be destroyed?

Reply:
The catch is required for this particular scenario. I've removed it but I will add it back

```
JSG_REQUIRE(!isSecureSocket, TypeError, "Cannot startTls on a TLS socket.");
```